Testscripts/Linux/LSVMBUS.sh

```diff
@@ -147,48 +148,51 @@ else
-    *Standard_L*s_v2*)
-        if [ "$VCPU" -gt 64 ];then
-            expected_scsi_counter=64
-        else
-            expected_scsi_counter=$VCPU
-        fi
-    ;;
-    # Default: the cpu count that attached to the SCSI driver is MIN(the number of vCPUs/4, 64).
-    *)
-        if [ "$VCPU" -gt $((64*4)) ];then
-            expected_scsi_counter=64
-        else
-            expected_scsi_counter=$(bc <<<"scale=2;$VCPU/4")
-            if ! [[ "$expected_scsi_counter" =~ \.00 ]]; then
-                # In this case we have a float number that needs to be rounded up.
-                # For example with 6 cores, scsi controller is spread on 2 cores.
-                expected_scsi_counter=$(bc <<<"("$expected_scsi_counter"+1)/1")
-            fi
-            # normalizing the number to integer
-            expected_scsi_counter=${expected_scsi_counter%.*}
-        fi
-    ;;
-esac
+# Default: the cpu count that attached to the SCSI driver is MIN(the number of vCPUs/4, 64).
+if [ "$VCPU" -gt $((64*4)) ];then
+    expected_scsi_counter=64
+else
+    expected_scsi_counter=$(bc <<<"scale=2;$VCPU/4")
+    if ! [[ "$expected_scsi_counter" =~ \.00 ]]; then
+        # In this case we have a float number that needs to be rounded up.
+        # For example with 6 cores, scsi controller is spread on 2 cores.
+        expected_scsi_counter=$(bc <<<"("$expected_scsi_counter"+1)/1")
+    fi
+    # normalizing the number to integer
+    expected_scsi_counter=${expected_scsi_counter%.*}
+fi
+
+testfail=0
+for adapter in "${!ADAPTER_DICT[@]}"; do
+    if [ "${ADAPTER_DICT[$adapter]}" != "$expected_network_counter" ]; then
+        error_msg="Error: values are wrong. Expected for network adapter $adapter channels: $expected_network_counter and actual: ${ADAPTER_DICT[$adapter]}"
+        LogErr "$error_msg"
+        UpdateSummary "$error_msg"
+        testfail=1
+    else
+        UpdateSummary "Expected for network adapter $adapter channels: $expected_network_counter and actual: ${ADAPTER_DICT[$adapter]}"
+    fi
+done
+
+for scsi in "${!SCSI_DICT[@]}"; do
+    if [ "${SCSI_DICT[$scsi]}" != "$expected_scsi_counter" ]; then
+        error_msg="Error: values are wrong. Expected for scsi $scsi channels: $expected_scsi_counter and actual: ${SCSI_DICT[$scsi]}"
+        LogErr "$error_msg"
+        UpdateSummary "$error_msg"
+        testfail=1
+    else
+        UpdateSummary "Expected for scsi $scsi channels: $expected_scsi_counter and actual: ${SCSI_DICT[$scsi]}"
+    fi
+done
+
 
-if [ "$network_counter" != "$expected_network_counter" ] || [ "$scsi_counter" != "$expected_scsi_counter" ]; then
-    error_msg="Error: values are wrong. Expected for network adapter: ${expected_network_counter} and actual: $network_counter;
-    expected for scsi controller: ${expected_scsi_counter}, actual: $scsi_counter."
+if [ "$testfail" == 1 ]; then
+    error_msg="scsi channels or network adapter channels are NOT expected, test fail."
     LogErr "$error_msg"
```

Comment:
seems we can ignore Log this Error (previous error messages have been logged with more clear information),  bug just UpdateSummary and SetTestStateFailed